Doc/library/functions.rst

```diff
@@ -1423,37 +1423,59 @@ are always available.  They are listed here in alphabetical order.
    :func:`codecs.register_error` is also valid.  The standard names
    include:
 
-   * ``'strict'`` to raise a :exc:`ValueError` exception if there is
-     an encoding error.  The default value of ``None`` has the same
-     effect.
-
-   * ``'ignore'`` ignores errors.  Note that ignoring encoding errors
-     can lead to data loss.
-
-   * ``'replace'`` causes a replacement marker (such as ``'?'``) to be inserted
-     where there is malformed data.
-
-   * ``'surrogateescape'`` will represent any incorrect bytes as low
-     surrogate code units ranging from U+DC80 to U+DCFF.
-     These surrogate code units will then be turned back into
-     the same bytes when the ``surrogateescape`` error handler is used
-     when writing data.  This is useful for processing files in an
-     unknown encoding.
-
-   * ``'xmlcharrefreplace'`` is only supported when writing to a file.
-     Characters not supported by the encoding are replaced with the
-     appropriate XML character reference :samp:`&#{nnn};`.
-
-   * ``'backslashreplace'`` replaces malformed data by Python's backslashed
-     escape sequences.
-
-   * ``'namereplace'`` (also only supported when writing)
-     replaces unsupported characters with ``\N{...}`` escape sequences.
+   .. list-table::
+      :header-rows: 1
+
+      * - Error handler
+        - Description
+      * - ``'strict'``
+        - Raise a :exc:`UnicodeError` (or a subclass) exception if there is
+          an error.  The default value of ``None`` has the same effect.
+      * - ``'ignore'``
+        - Ignore the malformed data and continue without further notice.
+          Note that ignoring encoding errors can lead to data loss.
+      * - ``'replace'``
+        - Replace malformed data with a replacement marker.
+          On encoding, use ``?`` (ASCII character).
+          On decoding, use ``�`` (U+FFFD, the official REPLACEMENT CHARACTER)
+      * - ``'backslashreplace'``
+        - Replace malformed data with backslashed escape sequences.
+          On encoding, use hexadecimal form of Unicode code point with formats
+          :samp:`\\x{hh}` :samp:`\\u{xxxx}` :samp:`\\U{xxxxxxxx}`.
+          On decoding, use hexadecimal form of byte value with format :samp:`\\x{hh}`.
+      * - ``'surrogateescape'``
+        - Will represent any incorrect bytes as low
+          surrogate code units ranging from ``U+DC80`` to ``U+DCFF``.
+          These surrogate code units will then be turned back into
+          the same bytes when the ``'surrogateescape'`` error handler is used
+          when writing data.  This is useful for processing files in an
+          unknown encoding.
+      * - ``'surrogatepass'``
+        - Only available for Unicode codecs.
+          Allow encoding and decoding surrogate code point
+          (``U+D800`` - ``U+DFFF``) as normal code point.  Otherwise these codecs
+          treat the presence of surrogate code point in :class:`str` as an error.
```

Comment:
```suggestion
          Allow encoding and decoding surrogate code points
          (``U+D800`` - ``U+DFFF``) as normal code points.  Otherwise these codecs
          treat the presence of surrogate code points in :class:`str` as an error.
```